src/foam/net/ip/FetchIPAddressInfo.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * NANOPAY CONFIDENTIAL
+ *
+ * [2020] nanopay Corporation
+ * All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of nanopay Corporation.
+ * The intellectual and technical concepts contained
+ * herein are proprietary to nanopay Corporation
+ * and may be covered by Canadian and Foreign Patents, patents
+ * in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from nanopay Corporation.
+ */
+
+ foam.CLASS({
+  package: 'foam.net.ip',
+  name: 'FetchIPAddressInfo',
+  extends: 'foam.dao.ReadOnlyDAO',
+  flags: ['java'],
+
+  documentation: `
+    Fetches IP Address Info from provider defined on the IP Address Info object.
+  `,
+
+  javaImports: [
+    'java.net.URI',
+    'java.net.http.HttpClient',
+    'java.net.http.HttpRequest',
+    'java.net.http.HttpRequest.BodyPublishers',
+    'java.net.http.HttpResponse.BodyHandlers',
+    'foam.net.IPSupport',
+    'foam.core.FObject',
+    'foam.net.ip.IPAddressInfo',
+    'foam.core.Detachable',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.dao.ProxySink',
+    'foam.dao.Sink',
+    'foam.nanos.auth.Subject',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'find_',
+      javaCode: `
+        FObject obj = getDelegate().find_(x, id);
+        IPAddressInfo ipInfo = null;
+        if ( obj == null ) {
+          ipInfo = (IPAddressInfo) ((ArraySink) getDelegate().select(new ArraySink())).getArray().get(0);
```

Comment:
It would be a lot more efficient to just do a find_(x. MLang.TRUE). This code will traverse the whole database since you didn't do a limit(1).

Reply:
If you had called select() instead of getDelegate().select() then you would have gotten the fetchInfo() behaviour you implemented below.